base_layer/core/src/mempool/service/service.rs

```diff
@@ -161,44 +176,151 @@ impl MempoolService {
-                );
-                if let Err(err) = self
-                    .network
-                    .report_gossip_message_validation_result(
-                        msg.message_id,
-                        msg.propagation_source,
-                        gossipsub::MessageAcceptance::Reject,
-                    )
-                    .await
-                {
-                    warn!(target: LOG_TARGET, "handle_incoming_tx call network: {err}");
+        let propagation_source = msg.propagation_source;
+        let proto::mempool::NewTransaction { payload: Some(payload) } = msg.message else {
+            warn!(target: LOG_TARGET, "Rejecting gossiped message from peer {propagation_source} because it contains an empty payload.");
+            if let Err(err) = self
+                .network
+                .report_gossip_message_validation_result(
+                    msg.message_id,
+                    propagation_source,
+                    gossipsub::MessageAcceptance::Reject,
+                )
+                .await
+            {
+                warn!(target: LOG_TARGET, "handle_incoming_tx call network: {err}");
+            }
+            return;
+        };
+
+        let transaction = match payload {
+            proto::mempool::new_transaction::Payload::ExcessSig(excess_sig) => {
+                match RistrettoSecretKey::from_canonical_bytes(&excess_sig) {
+                    Ok(excess_sig) => {
+                        let tx_id = MempoolTransactionId::try_from(excess_sig.as_bytes()).unwrap();
+                        match self.inbound_handlers.mempool().has_tx_with_excess_sig(excess_sig).await {
+                            Ok(status) if status.is_new() => {
+                                if self
+                                    .new_transactions_tx
+                                    .send(NewTransactionNotification {
+                                        transaction_id: tx_id,
+                                        propagation_source,
+                                        message_id: msg.message_id,
+                                    })
+                                    .is_err()
+                                {
+                                    error!(target: LOG_TARGET, "🚨 Want list sender has closed");
+                                }
+                            },
+                            Ok(status) if status.is_stored() => {
+                                // Already stored, we can forward the gossip
+                                if let Err(err) = self
+                                    .network
+                                    .report_gossip_message_validation_result(
+                                        msg.message_id,
+                                        msg.propagation_source,
+                                        gossipsub::MessageAcceptance::Accept,
```

Comment:
I think this should be ignore as well. 
The reason for this is, its a message about a tx that we already saw and processed. This means we already notified all peers we are connected to about this tx. So why send them another message about the tx?